src/interfaces/ApiActions.ts

```diff
@@ -0,0 +1,12 @@
+import Resource from './Resource'
+
+interface ApiActions {
+    get: (uriOrEntity: string | Resource, forceReload?: boolean) => Resource
+    reload: (uriOrEntity: string | Resource) => Promise<Resource>
+    post: (uriOrEntity: string | Resource, data: unknown) => Promise<Resource>
+    patch: (uriOrEntity: string | Resource, data: unknown) => Promise<Resource>
```

Comment:
Could we require `data` to be of `object` type?

Reply:
I had this before, then eslint complained:

```
Don't use `object` as a type. The `object` type is currently hard to use ([see this issue](https://github.com/microsoft/TypeScript/issues/21732)).
Consider using `Record<string, unknown>` instead, as it allows you to more easily inspect and use the keys.eslint@typescript-eslint/ban-types)
```

Which made me thinking if we really want to limit axios payload to objects? Axios has `any` as payload type (which actually is not encouraged by Typescript). At the moment, we may only post/patch JSON objects, but maybe someone wants to send binary data to the server, too. AFAIK HAL JSON only specifies the receiving part, but says nothing about the format when sending payload.